src/libs/utils.ts

```diff
@@ -22,11 +22,24 @@ export function formatDate(date: string | number) {
     return 'N/A';
   }
 
-  return new Date(date).toLocaleString('en-US', {
-    timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
-    timeStyle: 'short',
-    dateStyle: 'medium',
-  });
+  let formattedDate = '';
```

Comment:
there might be opportunities for code reuse here. Something like this might be more concise:
```
    const adjustedDate = typeof date === 'number' ? date * 1000 : date;
    return new Date(adjustedDate).toLocaleString('en-US', {
      timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
      timeStyle: 'short',
      dateStyle: 'medium',
    });
```